src/components/queryBuilder/QueryBuilder.tsx

```diff
@@ -40,8 +41,21 @@ export const QueryBuilder = (props: QueryBuilderProps) => {
           fields.push({ name: '*', label: 'ALL', type: 'string', picklistValues: [] });
           setBaseFieldsList(fields);
 
+          const dateTimeFields = fields.filter((f) => isDateTimeType(f.type));
+          if (dateTimeFields.length > 0) {
```

Comment:
This works right now however if you remove all the filters you can't get our new filter back.

Instead of adding to the filter list here, what if we toggle some state that says we have a dateTimeField? Then inside `onFiltersChange` we can reintroduce the filter whenever filters are removed completely?

Reply:
> if you remove all the filters you can't get our new filter back.

That was the exact idea - a user is proposed with a predefined filter once, but they are free to delete it entirely, and they won't be prompted with it again.

If we constantly push the predefined filter when filters are empty, it will result in the situation when the user deletes the filter, it just appears again immediately (i.e. it is not possible to delete it), so I wanted to explicitly avoid this behaviour.

Please correct me if I am getting this wrong.

Reply:
Sorry for the delay on this. I was misunderstanding how you wanted this to work. I'm good with this how it is and I'll approve right away.